### PR TITLE
Centralise all logging in Chucker

### PR DESCRIPTION
**Pull request by MiSikora**
## :page_facing_up: Context

Some parts of the code used `printStackTrace()` to log failures. I wanted to centralise logging point that can be used in Chucker. This also gives an opportunity to let users set an external logger in the future.

## :pencil: Changes

I changed `Logger` to an interface and made an instance of it available in `Chucker` object. Companion object of `Logger` delegates then to this property.

I also added a test extension that sets no-op logger. It is needed as we can't have a centralised logger that is not static due to logging in Android components as well.

One more thing – I removed "Chucker" from log messages. It is present in the tag anyway.

## :paperclip: Related PR

#452

## :no_entry_sign: Breaking

No.

## :hammer_and_wrench: How to test

Nothing special to check. Rely on unit tests.

## :stopwatch: Next steps

N/A

### CODE REVIEW

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt
Comment (cortinico) [inline library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt:187]: Should we generate an `Exception` here and call `Logger.error` instead?

**Review by cortinico — APPROVED** [review]
Sorry for the late review but I was 🎅 -ing. 
LGTM other than a minor comment 👍

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt
Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt:187]: I removed `Exception` because I thought it was used only to avoid `println` as there was no access here to a logger. I'm fine either way and can bring it back if you think it helps.

Comment (cortinico) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt:187]: I'm unsure about the value added tbh. I'm fine either way 👍

Comment (MiSikora) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt:187]: Ok, I'll leave it them. Just will fix a typo that I just noticed.

**Comment by vbuberen** [comment]
Wanted to take part in review as well 😕

**Comment by MiSikora** [comment]
Sorry, I didn't think this to be an important change to require review from everyone. If you have any comments leave them and I'll make a subsequent PR. :)

**Comment by vbuberen** [comment]
Ok, no problem. I thought that changes in classes with our public API would require more maintainers to check.

**Pull request merged**
